shoop/admin/modules/orders/views/shipment.py

```diff
@@ -17,15 +19,71 @@
 from shoop.admin.toolbar import PostActionButton, Toolbar
 from shoop.admin.utils.forms import add_form_errors_as_messages
 from shoop.admin.utils.urls import get_model_url
+from shoop.apps.provides import get_provide_objects
 from shoop.core.excs import NoProductsToShipException
 from shoop.core.models import Order, Product, Supplier
+from shoop.core.utils.shipments import generate_shipment_key
+
+FORM_MODIFIER_PROVIDER_KEY = "admin_extend_create_shipment_form"
+
+
+class ShipmentFormModifier(six.with_metaclass(abc.ABCMeta)):
+    @abc.abstractmethod
+    def get_extra_fields(self, order):
+        """
+        Extra fields for shipment creation form
+
+        :param order: Order linked to form
+        :type order: shoop.core.models.Order
+        :return: List of extra fields that should be added to form.
+        Tuple should contain field name and Django form field.
+        :rtype: list(tuple)|None
```

Comment:
Should return just list always, no need to return None ever, since empty list is OK too. Also, specify the return type of the tuple components too, i.e.
```
:rtype: list[(str,django.forms.Field)]
```